bundles/statistics/statsgrid2016/components/Legend.js

```diff
@@ -58,63 +50,105 @@ Oskari.clazz.define('Oskari.statistics.statsgrid.Legend', function(sandbox, loca
             return;
         }
         this._renderState.inProgress = true;
-        // start rendering
+
         var me = this;
-        var container = this.__element;
+        var container = this._element;
         var accordion = this._accordion;
-        // cleanup previous UI
         // NOTE! detach classification before re-render to keep eventhandlers
         this.editClassification.getElement().detach();
         accordion.removeAllPanels();
-        container.empty();
-        if(el) {
+
+        if ( container.find('.legend-noactive') ) {
+            container.find('.legend-noactive').remove();
+        }
+
+        for (var i = 0; i < container.children().length; i++) {
+            jQuery( container.children()[i] ).empty();
+        }
```

Comment:
Is there some reason that container.empty() cannot be used and instead you opt to clean up by looping through everything that it holds?